accname/name/comp_text_node.html

```diff
@@ -13,23 +13,96 @@
+<button class="ex" data-expectedlabel="button label" data-testname="button with text/comment/text nodes, with space">
+  button
+  <!-- comment node between text nodes with leading/trailing whitespace -->
+  label
+</button>
+<h3 class="ex" data-expectedlabel="heading label" data-testname="heading with text/comment/text nodes, with space">
   heading
   <!-- comment node between text nodes with leading/trailing whitespace -->
   label
-</h2>
+</h3>
+<a href="#" class="ex" data-expectedlabel="link label" data-testname="link with text/comment/text nodes, with space">
+  link
+  <!-- comment node between text nodes with leading/trailing whitespace -->
+  label
+</a>
+<br>
+
+<h1>text node, with tab char</h1>
+<button class="ex" data-expectedlabel="button label" data-testname="button with text node, with tab char">button  label</button>
+<h3 class="ex" data-expectedlabel="heading label" data-testname="heading with text node, with tab char">heading label</h3>
+<a href="#" class="ex" data-expectedlabel="link label" data-testname="link with text node, with tab char">link	label</a>
+<br>
+
+<h1>text node, with non-breaking space</h1>
+<button class="ex" data-expectedlabel="button&nbsp;label" data-testname="button with text node, with non-breaking space">button&nbsp;label</button>
+<h3 class="ex" data-expectedlabel="heading&nbsp;label" data-testname="heading with text node, with non-breaking space">heading&nbsp;label</h3>
+<a href="#" class="ex" data-expectedlabel="link&nbsp;label" data-testname="link with text node, with non-breaking space">link&nbsp;label</a>
+<br>
+
+<h1>text node, with extra non-breaking space</h1>
+<button class="ex" data-expectedlabel="button&nbsp;&nbsp;&nbsp;label" data-testname="button with text node, with extra non-breaking space">button&nbsp;&nbsp;&nbsp;label</button>
+<h3 class="ex" data-expectedlabel="heading&nbsp;&nbsp;&nbsp;label" data-testname="heading with text node, with extra non-breaking space">heading&nbsp;&nbsp;&nbsp;label</h3>
+<a href="#" class="ex" data-expectedlabel="link&nbsp;&nbsp;&nbsp;label" data-testname="link with text node, with extra non-breaking space">link&nbsp;&nbsp;&nbsp;label</a>
+<br>
+
+<h1>text node, with leading/trailing non-breaking space</h1>
+<button class="ex" data-expectedlabel="button&nbsp;label" data-testname="button with text node, with leading/trailing non-breaking space">&nbsp;button&nbsp;label&nbsp;</button>
+<h3 class="ex" data-expectedlabel="heading&nbsp;label" data-testname="heading with text node, with leading/trailing non-breaking space">&nbsp;heading&nbsp;label&nbsp;</h3>
+<a href="#" class="ex" data-expectedlabel="link&nbsp;label" data-testname="link with text node, with leading/trailing non-breaking space">&nbsp;link&nbsp;label&nbsp;</a>
+<br>
+
+<h1>text node, with leading/trailing non-breaking space</h1>
+<button class="ex" data-expectedlabel="button &nbsp; label" data-testname="button with text node, with mixed space and non-breaking space">button   &nbsp;   label</button>
```

Comment:
Thanks for writing all these tests!

I wasn't 100% sure how regular whitespace should be interacting with non-breaking spaces when I made the [Blink PR](https://github.com/chromium/chromium/commit/68243ce9b062184744635e3d75a39d429845ee7e), but it seems for now it should conform with the expectations you had here.

I was curious whether we should collapse `&nbsp;` and regular U+0020 spaces together into a single U+0020 space, but I wasn't able to find much guidance. I tried to look to the WHATWG Infra standard that seems to [define what it means to "collapse ASCII whitespace"](https://infra.spec.whatwg.org/#strip-and-collapse-ascii-whitespace),  but interestingly non-breaking spaces aren't included in the definition of ASCII whitespace in that spec. So I decided to leave that area alone for now.

Reply:
Yes, I'm thankful @zcorpan pointed that out earlier in the review, and I've updated to match the HTML definition of "ASCII Whitespace" that ARIA/AccName references.

Reply:
More discussion in [AccName #208](https://github.com/w3c/accname/issues/208).